src/cheapestTree.ts

```diff
@@ -101,3 +107,49 @@ function disableCraftForItemIds(
 
   return tree
 }
+
+function applyEfficiencyTiersToTree(
+  tree: NestedRecipe,
+  userEfficiencyTiers: Record<string, string>
+): NestedRecipe {
+  if (
+    ['102306', '102205', '103049'].includes(tree.id.toString()) &&
+    tree.merchant &&
+    tree.merchant.name.includes('Homestead Refinement')
+  ) {
+    const efficiencyTier = Number(userEfficiencyTiers[tree.id.toString()])
+    if (efficiencyTier > 0) {
+      const component = { ...tree.components[0] }
+      component.quantity = component.quantity / (efficiencyTier * 2)
+
+      if (component.id === 12142) {
```

Comment:
could you add the same comments here as in the core code? those really helped with understanding why the calculations work this way